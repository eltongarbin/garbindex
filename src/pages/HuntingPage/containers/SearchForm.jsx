import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { InputAdornment, IconButton, TextField } from '@material-ui/core';
import { Search as SearchIcon } from '@material-ui/icons';
import styled from 'styled-components';
import { connect } from 'react-redux';

import { actions } from '../state';

const TextFieldStyled = styled(TextField)`
  && {
    background-color: white;
  }
`;

class SearchForm extends PureComponent {
  state = { searchText: '' };

  handleInputSeachChange = (event) => {
    this.setState({ searchText: event.currentTarget.value.toLowerCase() });
  };

  handelSubmit = (event) => {
    event.preventDefault();
    this.props.searchForPokemon(this.state.searchText);
  };

  render() {
    const { searchText } = this.state;

    return (
      <form onSubmit={this.handelSubmit} autoComplete="off">
        <TextFieldStyled
          label="Name or Number"
          value={searchText}
          onChange={this.handleInputSeachChange}
          margin="normal"
          variant="outlined"
          type="text"
          fullWidth
          required
          InputLabelProps={{
            shrink: true
          }}
          autoFocus
          InputProps={{
            endAdornment: (
              <InputAdornment position="end">
                <IconButton
                  type="submit"
                  aria-label="Toggle password visibility"
                >
                  <SearchIcon />
                </IconButton>
              </InputAdornment>
            )
          }}
        />
      </form>
    );
  }
}

SearchForm.propTypes = {
  searchForPokemon: PropTypes.func.isRequired
};

const mapDispatchToProps = {
  searchForPokemon: actions.searchForPokemon.request
};

export default connect(
  null,
  mapDispatchToProps
)(SearchForm);
