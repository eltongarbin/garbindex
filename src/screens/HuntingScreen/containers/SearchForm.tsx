import React, { useState, useCallback } from 'react';
import { InputAdornment, IconButton, TextField } from '@material-ui/core';
import { Search as SearchIcon } from '@material-ui/icons';
import styled from 'styled-components';
import { useDispatch } from 'react-redux';

import { actions } from '../state';

const TextFieldStyled = styled(TextField)`
  && {
    background-color: white;
  }
`;

const SearchForm = React.memo(() => {
  const [searchText, setSearchText] = useState('');
  const dispatch = useDispatch();

  const handelSubmit = useCallback(
    (event) => {
      event.preventDefault();
      searchText && dispatch(actions.searchForPokemon.request(searchText));
    },
    [dispatch, searchText]
  );

  return (
    <form onSubmit={handelSubmit} autoComplete="off">
      <TextFieldStyled
        label="Name or Number"
        value={searchText}
        onChange={(event) => setSearchText(event.target.value)}
        margin="normal"
        variant="outlined"
        type="text"
        fullWidth
        required
        InputLabelProps={{
          shrink: true
        }}
        autoFocus
        InputProps={{
          endAdornment: (
            <InputAdornment position="end">
              <IconButton type="submit" aria-label="Search">
                <SearchIcon />
              </IconButton>
            </InputAdornment>
          )
        }}
        inputProps={{
          'data-testid': 'input-search'
        }}
      />
    </form>
  );
});

export default SearchForm;
