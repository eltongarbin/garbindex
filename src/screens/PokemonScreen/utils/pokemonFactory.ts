const defaultPokemon = {
  height: 6,
  id: 35,
  name: 'clefairy',
  sprites: {
    front_default:
      'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/35.png'
  },
  stats: [
    {
      base_stat: 35,
      name: 'speed',
      id: 6
    },
    {
      base_stat: 65,
      name: 'special-defense',
      id: 5
    },
    {
      base_stat: 60,
      name: 'special-attack',
      id: 4
    },
    {
      base_stat: 48,
      name: 'defense',
      id: 3
    },
    {
      base_stat: 45,
      name: 'attack',
      id: 2
    },
    {
      base_stat: 70,
      name: 'hp',
      id: 1
    }
  ],
  weight: 75,
  image:
    'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/35.png',
  abilitiesById: {
    '56': {
      id: 56,
      name: 'cute-charm'
    },
    '98': {
      id: 98,
      name: 'magic-guard'
    },
    '132': {
      id: 132,
      name: 'friend-guard'
    }
  },
  typesById: {
    '18': {
      id: 18,
      name: 'fairy'
    }
  },
  evolvedFrom: 'cleffa'
};

export function create(overrides: any = defaultPokemon) {
  return { ...defaultPokemon, ...overrides };
}
