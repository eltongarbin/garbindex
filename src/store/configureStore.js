import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { composeWithDevTools } from 'redux-devtools-extension';
import { persistStore, persistReducer } from 'redux-persist';
import storageSession from 'redux-persist/lib/storage/session';

import rootReducer from './rootReducer';
import rootSaga from './rootSaga';

const persistConfig = {
  key: 'root',
  storage: storageSession,
  whitelist: ['entities']
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

export default function configureStore(preloadedState) {
  const composeEnhancers = composeWithDevTools({});
  const sagaMiddleware = createSagaMiddleware();
  const store = createStore(
    persistedReducer,
    preloadedState,
    composeEnhancers(applyMiddleware(sagaMiddleware))
  );
  const persistor = persistStore(store);

  if (process.env.NODE_ENV !== 'production' && module.hot) {
    module.hot.accept('./rootReducer', () =>
      store.replaceReducer(persistedReducer)
    );
  }

  sagaMiddleware.run(rootSaga);

  return { store, persistor };
}
