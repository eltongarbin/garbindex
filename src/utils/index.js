export { default as history } from './history';
export { default as theme } from './theme';
