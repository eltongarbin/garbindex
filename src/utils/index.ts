export { default as history } from './history';
export { default as theme } from './theme';

export const extractParamId = (url: any) => {
  const [id] = url.match(/(\d+)(?!.*\d)/);
  return id;
};
