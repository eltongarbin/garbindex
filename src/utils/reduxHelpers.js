import { createAction } from 'redux-actions';
import { call, put } from 'redux-saga/effects';
import isEmpty from 'lodash.isempty';

export const createType = (stateKey, type) =>
  `${process.env.REACT_APP_NAME}/${stateKey}/${type}`;

export const createAsyncTypes = (stateKey, type) => ({
  REQUEST: `${createType(stateKey, type)}_REQUEST`,
  SUCCESS: `${createType(stateKey, type)}_SUCCESS`,
  FAILURE: `${createType(stateKey, type)}_FAILURE`
});

export const createAsyncActions = (asyncTypes) => ({
  request: createAction(asyncTypes.REQUEST),
  receive: createAction(asyncTypes.SUCCESS),
  error: createAction(asyncTypes.FAILURE)
});

export function* genericAsyncResolver(
  fnService,
  parameter,
  asyncActions,
  cbFormatData
) {
  try {
    const data = yield call(fnService, parameter);

    let dataFormatted;
    if (cbFormatData && !isEmpty(data)) {
      dataFormatted = yield cbFormatData(data);
    }

    yield put(asyncActions.receive(dataFormatted || data));
  } catch (error) {
    yield put(asyncActions.error(error));
  }
}
