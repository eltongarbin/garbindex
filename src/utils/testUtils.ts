/* eslint-disable react/forbid-foreign-prop-types */
import checkPropTypes from 'check-prop-types';

export const checkProps = (component, conformingProps) => {
  const propError = checkPropTypes(
    component.propTypes,
    conformingProps,
    'prop',
    component.name
  );

  expect(propError).toBeUndefined();
};
