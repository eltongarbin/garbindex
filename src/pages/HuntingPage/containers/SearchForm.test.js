import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { checkProps } from 'utils/testUtils';
import SearchForm from './SearchForm';
import configureStore from 'store';

const { store } = configureStore();
const setup = () => {
  const wrapper = shallow(<SearchForm store={store} />).dive();
  return wrapper;
};

describe('<SearchForm />', () => {
  it('matches the snapshot', () => {
    const tree = shallow(<SearchForm store={store} />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('does not throw warning with expected props', () => {
    checkProps(SearchForm, { searchForPokemon: () => {} });
  });

  it('`searchForPokemon` action creator is a function prop', () => {
    const wrapper = setup();
    const searchForPokemonProp = wrapper.instance().props.searchForPokemon;
    expect(searchForPokemonProp).toBeInstanceOf(Function);
  });
});
