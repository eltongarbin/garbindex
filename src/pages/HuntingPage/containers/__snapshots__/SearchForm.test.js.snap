// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchForm /> matches the snapshot 1`] = `
<SearchForm
  searchForPokemon={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
/>
`;
